pkg/persistence/config/loader/config_loader_test.go

```diff
@@ -1167,6 +1169,48 @@ configs:
 `,
 			wantErrorsContain: []string{"unknown API: bucket"},
 		},
+		{
+			name:             "grail filter-segment config with FF on",
+			envVars:          map[string]string{featureflags.Temporary[featureflags.GrailFilterSegment].EnvName(): "true"},
+			filePathArgument: "test-file.yaml",
+			filePathOnDisk:   "test-file.yaml",
+			fileContentOnDisk: `
+configs:
+- id: profile-id
+  config:
+    template: 'profile.json'
+  type: filter-segment
+`,
+			wantConfigs: []config.Config{
+				{
+					Coordinate: coordinate.Coordinate{
+						Project:  "project",
+						Type:     "filter-segment",
+						ConfigId: "profile-id",
+					},
+					Type:        config.GrailFilterSegment{},
+					Template:    template.NewInMemoryTemplate("profile.json", "{}"),
+					Parameters:  config.Parameters{},
```

Comment:
```suggestion
					Template:    template.NewInMemoryTemplate("filter-segment.json", "{}"),
```

Reply:
I don't understand this suggestion. `Template` is already set in line 1192

Reply:
It is about renaming the name of the file...